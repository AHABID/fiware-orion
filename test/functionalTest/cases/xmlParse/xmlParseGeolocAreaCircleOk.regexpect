1: +++++++++++++++++++++++++++++++++++++
<queryContextResponse>
  <errorCode>
    <code>404</code>
    <reasonPhrase>No context element found</reasonPhrase>
  </errorCode>
</queryContextResponse>
2: +++++++++++++++++++++++++++++++++++++
<updateContextResponse>
  <contextResponseList>
    <contextElementResponse>
      <contextElement>
        <entityId type="Room" isPattern="false">
          <id>entity01</id>
        </entityId>
        <contextAttributeList>
          <contextAttribute>
            <name>temperature</name>
            <type>degree</type>
            <contextValue></contextValue>
          </contextAttribute>
          <contextAttribute>
            <name>lightstatus</name>
            <type>light</type>
            <contextValue></contextValue>
          </contextAttribute>
        </contextAttributeList>
      </contextElement>
      <statusCode>
        <code>200</code>
        <reasonPhrase>OK</reasonPhrase>
      </statusCode>
    </contextElementResponse>
  </contextResponseList>
</updateContextResponse>
3: +++++++++++++++++++++++++++++++++++++
<queryContextResponse>
  <contextResponseList>
    <contextElementResponse>
      <contextElement>
        <entityId type="Room" isPattern="false">
          <id>entity01</id>
        </entityId>
        <contextAttributeList>
          <contextAttribute>
            <name>temperature</name>
            <type>degree</type>
            <contextValue>11</contextValue>
          </contextAttribute>
          <contextAttribute>
            <name>lightstatus</name>
            <type>light</type>
            <contextValue>L23</contextValue>
          </contextAttribute>
        </contextAttributeList>
      </contextElement>
      <statusCode>
        <code>200</code>
        <reasonPhrase>OK</reasonPhrase>
      </statusCode>
    </contextElementResponse>
  </contextResponseList>
</queryContextResponse>
4: +++++++++++++++++++++++++++++++++++++
<subscribeContextResponse>
  <subscribeResponse>
    <subscriptionId>REGEX([0-9a-f]{24})</subscriptionId>
    <duration>P5Y</duration>
    <throttling>P5Y</throttling>
  </subscribeResponse>
</subscribeContextResponse>
5: +++++++++++++++++++++++++++++++++++++
circleRadius: Got a circleRadius: 10
circleRadius: Got a circleRadius: 10
circleRadius: Got a circleRadius: 11
6: +++++++++++++++++++++++++++++++++++++
