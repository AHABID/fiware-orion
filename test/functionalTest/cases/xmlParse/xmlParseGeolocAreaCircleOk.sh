source ../harnessFunctions.sh

ucr='<?xml version="1.0" encoding="UTF-8"?>
<updateContextRequest>
  <contextElementList>
    <contextElement>
      <entityId type="Room" isPattern="false">
        <id>entity01</id>
      </entityId>
      <contextAttributeList>
        <contextAttribute>
          <name>temperature</name>
          <type>degree</type>
          <contextValue>11</contextValue>
        </contextAttribute>
        <contextAttribute>
          <name>lightstatus</name>
          <type>light</type>
          <contextValue>L23</contextValue>
        </contextAttribute>
      </contextAttributeList>
    </contextElement>
  </contextElementList>
  <updateAction>APPEND</updateAction>
</updateContextRequest>'

qcr='<?xml version="1.0" encoding="UTF-8"?>
<queryContextRequest>
  <entityIdList>
    <entityId type="Room" isPattern="false">
      <id>entity01</id>
    </entityId>
  </entityIdList>
  <attributeList>
    <attribute>temperature</attribute>
    <attribute>occupancy</attribute>
    <attribute>lightstatus</attribute>
  </attributeList>
  <restriction>
    <attributeExpression>Attribute Expression</attributeExpression>
    <scope>
      <operationScope>
        <scopeType>st1</scopeType>
        <scopeValue>sv1</scopeValue>
      </operationScope>
      <operationScope>
        <scopeType>FIWARE_Location</scopeType>
        <scopeValue>
          <circle>
            <radius>10</radius>
            <centerLatitude>45</centerLatitude>
            <centerLongitude>45</centerLongitude>
            <inverted>0</inverted>
          </circle>
        </scopeValue>
      </operationScope>
      <operationScope>
        <scopeType>st2</scopeType>
        <scopeValue>sv2</scopeValue>
      </operationScope>
    </scope>
  </restriction>
</queryContextRequest>'

scr='<?xml version="1.0" encoding="UTF-8"?>
<subscribeContextRequest>
  <entityIdList>
    <entityId type="Room" isPattern="false">
      <id>ConferenceRoom</id>   
    </entityId>
    <entityId type="Room" isPattern="false">
      <id>OfficeRoom</id>   
    </entityId>
  </entityIdList>
  <attributeList>
    <attribute>temperature</attribute>
    <attribute>lightstatus</attribute>
  </attributeList>
  <reference>http://127.0.0.1:1028</reference>
  <duration>P5Y</duration>
  <restriction>
    <attributeExpression>testRestriction</attributeExpression>
    <scope>
      <operationScope>
        <scopeType>scope1</scopeType>
        <scopeValue>sval1</scopeValue>
      </operationScope>
      <operationScope>
        <scopeType>scope2</scopeType>
        <scopeValue>sval2</scopeValue>
      </operationScope>
      <operationScope>
        <scopeType>FIWARE_Location</scopeType>
        <scopeValue>
          <circle>
            <radius>11</radius>
            <centerLatitude>55</centerLatitude>
            <centerLongitude>55</centerLongitude>
            <inverted>0</inverted>
          </circle>
        </scopeValue>
      </operationScope>
    </scope>
  </restriction>
  <notifyConditions>
    <notifyCondition>
      <type>ONCHANGE</type>
      <condValueList>
        <condValue>temperature</condValue>
        <condValue>lightstatus</condValue>
      </condValueList>
      <restriction>restriction</restriction>
    </notifyCondition>
  </notifyConditions>
  <throttling>P5Y</throttling>
</subscribeContextRequest>'

ucsr='<?xml version="1.0" encoding="UTF-8"?>
<updateContextSubscriptionRequest>
  <duration>P50Y</duration>
  <restriction>
    <attributeExpression>AttriTest</attributeExpression>
    <scope>
      <operationScope>
        <scopeType>st1</scopeType>
        <scopeValue>sv1</scopeValue>
      </operationScope>
      <operationScope>
        <scopeType>FIWARE_Location</scopeType>
        <scopeValue>
          <circle>
            <radius>12</radius>
            <centerLatitude>65</centerLatitude>
            <centerLongitude>65</centerLongitude>
            <inverted>0</inverted>
          </circle>
        </scopeValue>
      </operationScope>
    </scope>
  </restriction>
  <subscriptionId>111222333444555666777888</subscriptionId>
  <notifyConditions>
    <notifyCondition>
      <type>ONCHANGE</type>
      <condValueList>
        <condValue>CondValue3</condValue>
        <condValue>CondValue4</condValue>
      </condValueList>
    </notifyCondition>
  </notifyConditions>
  <throttling>P5Y</throttling>
</updateContextSubscriptionRequest>'

echo '1: +++++++++++++++++++++++++++++++++++++'
curl localhost:${BROKER_PORT}/ngsi10/queryContext -s -S --header "Content-Type: application/xml" -d "$qcr" 
echo '2: +++++++++++++++++++++++++++++++++++++'
curl localhost:${BROKER_PORT}/ngsi10/updateContext -s -S --header "Content-Type: application/xml" -d "$ucr"
echo '3: +++++++++++++++++++++++++++++++++++++'
curl localhost:${BROKER_PORT}/ngsi10/queryContext -s -S --header "Content-Type: application/xml" -d "$qcr"
echo '4: +++++++++++++++++++++++++++++++++++++'
curl localhost:${BROKER_PORT}/ngsi10/subscribeContext -s -S --header "Content-Type: application/xml" -d "$scr"
echo '5: +++++++++++++++++++++++++++++++++++++'
grep 'circleRadius: Got a circleRadius' /tmp/contextBrokerLog | awk -F'] ' '{ print $2 }'
echo '6: +++++++++++++++++++++++++++++++++++++'

