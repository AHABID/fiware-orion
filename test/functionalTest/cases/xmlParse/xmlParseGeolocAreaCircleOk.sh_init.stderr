+ source ../harnessFunctions.sh
++ '[' YES '!=' YES ']'
+ dbInit CB
+ role=CB
+ '[' CB == CB ']'
+ echo 'db.dropDatabase()'
+ mongo testharness --quiet
+ brokerStart CB
+ role=CB
+ traceLevels=
+ ipVersion=
+ shift
+ shift
+ shift
+ extraParams=
+ '[' CB == '' ']'
+ '[' '' == '' ']'
+ traceLevels=0-255
+ localBrokerStop CB
+ role=CB
+ '[' CB == CB ']'
+ port=9999
++ ps -fe
++ grep 9999
++ wc -l
++ grep contextBroker
+ running_broker=0
+ '[' 0 -ne 0 ']'
+ localBrokerStart CB 0-255
+ role=CB
+ traceLevels=0-255
+ ipVersion=
+ shift
+ shift
+ shift
+ extraParams=
+ IPvOption=
+ '[' '' == IPV4 ']'
+ '[' '' == IPV6 ']'
+ '[' CB == CB ']'
+ port=9999
+ CB_START_CMD='contextBroker -vvvvv -harakiri -port 9999 -pidpath /tmp/orion_9999.pid     -db testharness -t 0-255  '
+ '[' '' == '' ']'
+ contextBroker -vvvvv -harakiri -port 9999 -pidpath /tmp/orion_9999.pid -db testharness -t 0-255
+ sleep 1
++ ps -fe
++ wc -l
++ grep contextBroker
++ grep 9999
+ running_broker=1
+ '[' 1 -ne 1 ']'
